src/hal/environment.cpp

```diff
@@ -129,48 +155,39 @@ void OswHal::Environment::setupStepStatistics() {
         if ((resetDay >= 0 && resetDay < 8) && (unsigned short) resetDay == currDoW + 1) // (e.g. 1 - 7 are days, 0 is disabled)
             this->_stepsSum = 0;
 
-        OSW_LOG_D("Updated steps from DoW ", lastDoW, " to DoW ", currDoW);
-        lastDoW = currDoW;
+        this->_stepsLastDoW = currDoW;
+        OSW_LOG_D("Updated steps from DoW ", this->_stepsLastDoW, " to DoW ", currDoW);
+
         // write step cache + stepsum
         res = prefs.putBytes(PREFS_STEPS_STATS, &this->_stepsCache, sizeof(this->_stepsCache)) == sizeof(this->_stepsCache);
         assert(res);
         res = prefs.putUInt(PREFS_STEPS_ALL, this->_stepsSum) == sizeof(this->_stepsSum);
         assert(res);
-        res = prefs.putUInt(PREFS_STEPS_DAYLASTCHECKED, lastDoW) == sizeof(lastDoW);
+        res = prefs.putUInt(PREFS_STEPS_DAYLASTCHECKED, this->_stepsLastDoW) == sizeof(this->_stepsLastDoW);
         assert(res);
+        prefs.end();
     }
-    prefs.end();
+
 #ifndef NDEBUG
-    String stepHistoryDbgMsg = "Current step history (day " + String(currDoW) + ", today " + String(OswHal::getInstance()->environment->getStepsToday()) + ", sum " + String(this->_stepsSum) + ") is: {";
-    for(size_t i = 0; i < 7; i++) {
-        if(i > 0)
-            stepHistoryDbgMsg += ", ";
-        if(i == currDoW)
-            stepHistoryDbgMsg += "[";
-        stepHistoryDbgMsg += this->_stepsCache[i];
-        if(i == currDoW)
-            stepHistoryDbgMsg += "]";
+    if(changedDoW or alwaysPrintStepStatistics) {
+        String stepHistoryDbgMsg = "Current step history (day " + String(currDoW) + ", today " + String(OswHal::getInstance()->environment->getStepsToday()) + ", sum " + String(this->_stepsSum) + ") is: {";
+        for(size_t i = 0; i < 7; i++) {
+            if(i > 0)
+                stepHistoryDbgMsg += ", ";
+            if(i == currDoW)
+                stepHistoryDbgMsg += "[";
+            stepHistoryDbgMsg += this->_stepsCache[i];
+            if(i == currDoW)
+                stepHistoryDbgMsg += "]";
+        }
+        stepHistoryDbgMsg += "}";
+        OSW_LOG_D(stepHistoryDbgMsg);
     }
-    stepHistoryDbgMsg += "}";
```

Comment:
Dbg....? Debug? how about `stepHistoryLog`?
Because abbreviated words and too long. Actually, It doesn't really matter.

Reply:
I mean the correct speaking name would be `stepHistoryDebugMessage`, but I hate that. Your proposal `stepHistoryLog` does not seem to be specific enough and may interfere with names down the road. Sooo... I don't care. If you really think this is bad, YKWTD.